### PR TITLE
Pass app and device fields to Prebid server

### PR DESCRIPTION
**Pull request by AntoineJac**
pbsrequest.go code is indeed developed to read the app fields present in the root of PBSRequest json not in the PBSRequest.ad_units child object. So far no app bundle is passed in the prebid server payload to the differents bidders...
With this fix, the device and app should be passed in the setConfig rather than AdUnits.

## Type of change

- [x] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

File: modules/prebidServerBidAdapter.js [hunk 1]
```diff
@@ -193,6 +193,8 @@ export function PrebidServer() {
   /* Prebid executes this function when the page asks to send out bid requests */
   baseAdapter.callBids = function(s2sBidRequest, bidRequests, addBidResponse, done, ajax) {
     const isDebug = !!getConfig('debug');
+    const app = !!getConfig('app');
```
Comment (mkendall07) [inline modules/prebidServerBidAdapter.js]: are pubs setting `app` and `device` in the config?

Comment (AntoineJac) [reply modules/prebidServerBidAdapter.js]: The use case here is using Postbids in an inapp environment via DFP. Nothing is explained on prebid about passing the app and device data yet so using the config seem the most relevant and logical way of doing it. 

On the server code (go) all is already done to be able to read this fields and use in requests to bidders.

Comment (mkendall07) [reply modules/prebidServerBidAdapter.js]: yeah that makes sense. If `app` and `device` are ORTB objects (and I think they are / will be soon) based on the new ORTB endpoint, then yes I think it makes sense to put these into the config as opposed to the S2S setting object. These can be used universally then vs just S2S.

**Comment by mkendall07** [comment]
@AntoineJac 
I think this is good but we need a docs update to merge. I'm thinking this page should probably have some info on the structure of these items:
https://github.com/prebid/prebid.github.io/blob/master/dev-docs/get-started-with-prebid-server.md

**Comment by mkendall07** [comment]
@bretg @mjacobsonny for thoughts as well.

**Comment by bretg** [comment]
It's unclear where this scenario should be documented, but honestly the main getting-started page doesn't seem right. I view this PostBid-through-PrebidServerBidAdapter as a one-off that few pubs will need.

How about a FAQ entry?

**Q: How can I use Prebid Server in a mobile app post-bid scenario?**

Yes. Just schedule a creative in the ad server that:

1) Loads Prebid JS
2) Sets up the AdUnit
3) Sets the app and device with setConfig along with the s2sConfig. e.g.
```
pbjs.setConfig({
    s2sConfig: {
    ...
    },
    app: "com.test.app",
    device: "WHAT_VALUES_DOES_THIS_TAKE?"
});
```

**Comment by AntoineJac** [comment]
Do you please have any ETA for this release to be deployed?

**Comment by mkendall07** [comment]
@AntoineJac 
I don't think we have any issues with the code - but we need a docs PR.
@bretg 
I'm good with the FAQ. 

Separately we should have a dedicated `setConfig` API page.

**Comment by bretg** [comment]
@AntoineJac and I are going to revise the interface to better reflect OpenRTB:

```
pbjs.setConfig({
    s2sConfig: {
    ...
    },
    app: {
        bundle: "com.test.app"
    },
    device: {
         ifa: "6D92078A-8246-4BA4-AE5B-76104861E7DC"
    }
});
```

Also, we think the 'app' and 'device' args for the PBS /auction endpoint are objects, not strings. Antoine is confirming, and then I'll submit a doc PR

**Comment by bretg** [comment]
doc PR submitted https://github.com/prebid/prebid.github.io/pull/588

**Comment by bretg** [comment]
doc PR open. back to you @mkendall07

**Comment by bretg** [comment]
This PR created a bug -- Prebid Server can't handle getting app and device parameters.

**Comment by matthewlane** [comment]
I just ran into this as well. I think it's because `device` and `app` are being converted to booleans (with `!!getConfig('app');`) rather than passed as objects. Dropping the boolean conversion may fix this

**Pull request merged**